### PR TITLE
Fix on previous PR + expanding support

### PR DESCRIPTION
**Pull request by adefossez**
## Checklist

- [x ] Read CONTRIBUTING.md, and accept the CLA by including the provided snippet. We will not accept PR without this.
- [x ] Run pre-commit hook.
- [x ] If you changed Rust code, run `cargo check`, `cargo clippy`, `cargo test`.

## PR Description

Removes some training code that are no longer needed (attr and cfg dropout).

Adding clone on the output of the graphed encoder and decoder to avoid getting burnt.

Adding a hook system to customize the LM output.

### CODE REVIEW

File: moshi/moshi/models/lm.py [hunk 1]
```diff
@@ -554,6 +563,7 @@ def _init_streaming_state(self, batch_size: int) -> _LMGenState:
             if state.condition_sum is not None:
                 assert state.condition_sum.shape[0] == batch_size, "cfg requires 2x more conditions."
             if state.condition_cross is not None:
+                torch.save(state.condition_cross, '/home/alex/tmp/dbg_cross.th')
```
Comment (LaurentMazare) [inline moshi/moshi/models/lm.py:566]: Remove this debug bit?

**Review by LaurentMazare — APPROVED** [review]
